ftw/mail/inbound.py

```diff
@@ -142,75 +131,6 @@ def createMailInContainer(container, message):
-    fti = getUtility(IDexterityFTI, name='ftw.mail.mail')
-    schema = fti.lookupSchema()
-    field_type = getFields(schema)['message']._type
-    message_value = field_type(data=message,
-                       contentType='message/rfc822', filename=u'message.eml')
-    # create mail object
-    content = createContent('ftw.mail.mail', message=message_value)
-
-    container = aq_inner(container)
-    container_fti = container.getTypeInfo()
-
-    # check permission
-    permission = queryUtility(IPermission, name='ftw.mail.AddInboundMail')
-    if permission is None:
-        raise Unauthorized("Cannot create %s" % content.portal_type)
-    if not getSecurityManager().checkPermission(permission.title, container):
-        raise Unauthorized("Cannot create %s" % content.portal_type)
-
-    # check addable types
-    if container_fti is not None and \
-            not container_fti.allowType(content.portal_type):
-        raise ValueError("Disallowed subobject type: %s" % (
-                content.portal_type))
-
-    normalizer = queryUtility(IIDNormalizer)
-    normalized_subject = normalizer.normalize(content.title)
-
-    name = INameChooser(container).chooseName(normalized_subject, content)
-    content.id = name
-
-    newName = container._setObject(name, content)
-    obj = container._getOb(newName)
-    obj = set_defaults(obj, container)
-    obj.reindexObject()
-    return obj
-
-
-def set_defaults(obj, container):
-    """set the default value for all fields on the mail object
-    (including additional behaviors)"""
-
-    for schema in iterSchemata(obj):
-        for name, field in getFieldsInOrder(schema):
-            # Remove acquisition wrapper when getting field value so
-            # determining if a field is already set works as expected
-            value = field.get(field.interface(aq_base(obj)))
-            if value == field.missing_value or value is None:
-                # bind the field for choices with named vocabularies
-                field = field.bind(obj)
-
-                # No value is set, so we try to set the default value
-                # otherwise we set the missing value
-                default = queryMultiAdapter((
-                        container,
-                        container.REQUEST,  # request
-                        None,  # form
-                        field,
-                        None,  # Widget
-                        ), IValue, name='default')
-                if default is not None:
-                    default = default.get()
-                if default is None:
-                    default = getattr(field, 'default', None)
-                if default is None:
-                    try:
-                        default = field.missing_value
-                    except:
-                        pass
-                field.set(field.interface(obj), default)
-    return obj
+    creator = getMultiAdapter((container, getRequest()),
```

Comment:
instead of `getRequest` you may use container.REQUEST. But this just saves a import statement, nothing more.